Riot/Modules/Room/Views/InputToolbar/RoomInputToolbarView.m

```diff
@@ -443,9 +362,63 @@ - (void)updateSendButtonWithMessage:(NSString *)textMessage
         self.rightInputToolbarButton.alpha = 0;
         self.messageComposerContainerTrailingConstraint.constant = 12;
     }
+    
     [self layoutIfNeeded];
 }
 
+#pragma mark - properties
+
+- (void)setActionMenuOpened:(BOOL)actionMenuOpened
+{
+    if (_actionMenuOpened != actionMenuOpened)
+    {
+        _actionMenuOpened = actionMenuOpened;
+        
+        if (self->growingTextView.internalTextView.selectedRange.length > 0)
+        {
+            NSRange range = self->growingTextView.internalTextView.selectedRange;
+            range.location = range.location + range.length;
+            range.length = 0;
+            self->growingTextView.internalTextView.selectedRange = range;
+        }
+
+        if (_actionMenuOpened) {
+            self.actionsBar.hidden = NO;
+            [self.actionsBar animateWithShowIn:_actionMenuOpened completion:nil];
+        }
+        else
+        {
+            [self.actionsBar animateWithShowIn:_actionMenuOpened completion:^(BOOL finished) {
+                self.actionsBar.hidden = YES;
+            }];
+        }
+        
+        [UIView animateWithDuration:.4 delay:0 usingSpringWithDamping:0.45 initialSpringVelocity:7 options:UIViewAnimationOptionCurveEaseIn animations:^{
```

Comment:
We could all use constants for these animation values. Also we could nullify the completion block if we don't use it.